pkg/plugin/utils.go

```diff
@@ -36,33 +36,38 @@ const (
 )
 
 func GetIpsByOnField(on string, context Context, phase Phase) (ipList []net.IP, err error) {
-	on = strings.TrimSpace(on)
-	if strings.Contains(on, EqualSymbol) {
-		if (phase != PhasePostInstall && phase != PhasePostJoin) && phase != PhasePreClean {
-			logrus.Warnf("Current phase is %s. When nodes is specified with a label, the plugin action must be PostInstall or PostJoin, ", phase)
-			return nil, nil
+	splits := strings.Split(on, SplitSymbol)
+	for _, split := range splits {
+		var ips []net.IP
+		split = strings.TrimSpace(split)
+		if strings.Contains(split, EqualSymbol) {
+			if (phase != PhasePostInstall && phase != PhasePostJoin) && phase != PhasePreClean {
+				logrus.Warnf("Current phase is %s. When nodes is specified with a label, the plugin action must be PostInstall or PostJoin, ", phase)
+				return nil, nil
+			}
+			client, err := k8s.Newk8sClient()
+			if err != nil {
+				return nil, fmt.Errorf("failed to get k8s client: %v", err)
+			}
+			ips, err = client.ListNodeIPByLabel(strings.TrimSpace(split))
+			if err != nil {
+				return nil, err
+			}
+		} else if split == common.MASTER || split == common.NODE {
+			ips = context.Cluster.GetIPSByRole(split)
+		} else if split == common.MASTER0 {
+			ips = context.Cluster.GetIPSByRole(common.MASTER)
+			if len(ips) < 1 {
+				return nil, fmt.Errorf("invalid on filed: [%s]", on)
+			}
+			ips = ips[:1]
+		} else {
+			ips = utilsnet.DisassembleIPList(split)
 		}
-		client, err := k8s.Newk8sClient()
-		if err != nil {
-			return nil, fmt.Errorf("failed to get k8s client: %v", err)
+		if len(ips) == 0 {
+			logrus.Debugf("node not found by on field [%s]", on)
```

Comment:
Still wish to return the detailed error, rather than just throwing out a debug log.

Reply:
oh, it would be better